contrib/babelfishpg_tsql/sql/sys_views.sql

```diff
@@ -1195,27 +1195,27 @@ select t.name,t.type, ns.oid as schemaid from
 ) t(name,schema_name, type)
 inner join pg_catalog.pg_namespace ns on t.schema_name = ns.nspname
 
-union all 
+union all
 
 -- This portion of view retrieves information on objects that reside in a schema in any number of databases.
 -- For example, 'dbo' schema can exist in the 'master', 'tempdb', 'msdb', and any user created database.
-select t.name,t.type, ns.oid  as schemaid from
+select t.name,t.type, ns.oid as schemaid from
 (
-  values 
+  values
     ('sysdatabases','dbo','V')
 ) t (name, schema_name, type)
-inner join sys.babelfish_namespace_ext b on t.schema_name=b.orig_name
+inner join sys.babelfish_namespace_ext b on t.schema_name = b.orig_name
 inner join pg_catalog.pg_namespace ns on b.nspname = ns.nspname;
 GRANT SELECT ON sys.shipped_objects_not_in_sys TO PUBLIC;
 
 create or replace view sys.all_objects as
 select 
-    cast (name as sys.sysname) 
+    cast (name as sys.sysname) collate sys.database_default
   , cast (object_id as integer) 
   , cast ( principal_id as integer)
   , cast (schema_id as integer)
   , cast (parent_object_id as integer)
-  , cast (type as char(2))
+  , cast (type as char(2)) collate sys.database_default
```

Comment:
Sorry, commenting here, because further down, I'm not able to comment, but it is related.

Should similar change be done in the sys.data_spaces view for this column?
CAST('FG' as CHAR(2)) AS type,  --> CAST('FG' as CHAR(2)) collate sys.database_default AS type,

Then subsequently the sys.filegroups view is using this column in where statement
FROM sys.data_spaces ds WHERE type = 'FG';

Does the above where statement need the collate as well?
FROM sys.data_spaces ds WHERE type = 'FG' collate sys.database_default;

Reply:
The datatype for type column for `sys.data_spaces` view is char ,which would not have `sys.database_default` collation by default (suggestion:we want to use sys.bpchar instead to inherit the default collation) because of which sys.filegroups need explicit collate clause